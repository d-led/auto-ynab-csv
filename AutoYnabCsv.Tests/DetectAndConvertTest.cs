using ApprovalTests;
using ApprovalTests.Reporters;
using AutoYnabCsv.Converters;
using AutoYnabCsv.Exporters;

namespace Tests;

[TestClass]
[UseReporter(typeof(DiffReporter))]
public class DetectAndConvertTest
{
    [TestMethod]
    [DataRow("n26-download.csv")]
    public void DetectKnownCsv(string inputFilename)
    {
        var input = TestHelpers.SampleTextOf(inputFilename);
        var conversion = DetectAndConvert.Instance.Convert(input);
        var csv = YnabCsvExporter.Export(conversion);
        Approvals.Verify(csv);
    }
}