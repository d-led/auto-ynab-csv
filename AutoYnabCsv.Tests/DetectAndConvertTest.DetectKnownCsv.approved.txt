Date,Payee,Memo,Inflow,Outflow
2024-12-12,Some bakery,,0,5.95
2024-12-13,Some sushi place,,0,14
2024-12-27,Somebody,Earnings,1234.54,0
